src/ray/object_manager/test/get_request_queue_test.cc

```diff
@@ -0,0 +1,129 @@
+  MOCK_METHOD1(SealObject, const LocalObject *(const ObjectID &object_id));
+  MOCK_METHOD1(AbortObject, flatbuf::PlasmaError(const ObjectID &object_id));
+  MOCK_METHOD1(DeleteObject, flatbuf::PlasmaError(const ObjectID &object_id));
+  MOCK_METHOD1(AddReference, bool(const ObjectID &object_id));
+  MOCK_METHOD1(RemoveReference, bool(const ObjectID &object_id));
+};
+struct GetRequestQueueTest : public Test {
+ public:
+  GetRequestQueueTest() : io_work_(io_context_) {}
+  void SetUp() override {
+    Test::SetUp();
+    thread_ = std::thread([this] { io_context_.run(); });
+  }
+
+  void TearDown() override {
+    io_context_.stop();
+    if (thread_.joinable()) {
+      thread_.join();
+    }
+  }
+
+ protected:
+  instrumented_io_context io_context_;
+  boost::asio::io_service::work io_work_;
+  std::thread thread_;
+};
+
+TEST_F(GetRequestQueueTest, TestAddRequest) {
+  MockObjectLifecycleManager object_lifecycle_manager;
+  GetRequestQueue get_request_queue(io_context_, object_lifecycle_manager);
+  auto client = std::make_shared<MockClient>();
+  ObjectID object_id = ObjectID::FromRandom();
+  std::vector<ObjectID> object_ids{object_id};
+  LocalObject object1{Allocation()};
+  object1.object_info.data_size = 10;
+  object1.object_info.metadata_size = 0;
+  /// Test object has been satisfied.
+  {
+    /// Mock the object already sealed.
+    object1.state = ObjectState::PLASMA_SEALED;
+    EXPECT_CALL(object_lifecycle_manager, GetObject(_))
+        .Times(1)
+        .WillOnce(Return(&object1));
+    bool satisfied = false;
+    get_request_queue.AddRequest(
+        client, object_ids, 1000, false,
+        [&](const std::shared_ptr<GetRequest> &get_req) { satisfied = true; });
+    EXPECT_TRUE(satisfied);
+  }
+
+  /// Test object not satisfied, time out.
+  {
+    object1.state = ObjectState::PLASMA_CREATED;
+    EXPECT_CALL(object_lifecycle_manager, GetObject(_))
+        .Times(1)
+        .WillOnce(Return(&object1));
+    std::promise<bool> promise;
+    get_request_queue.AddRequest(
+        client, object_ids, 1000, false,
+        [&](const std::shared_ptr<GetRequest> &get_req) { promise.set_value(true); });
+    promise.get_future().get();
+  }
+
+  get_request_queue.RemoveGetRequestsForClient(client);
+  EXPECT_FALSE(get_request_queue.IsGetRequestExist(object_id));
+
+  /// Test object not satisfied, then sealed.
+  {
+    object1.state = ObjectState::PLASMA_CREATED;
+    EXPECT_CALL(object_lifecycle_manager, GetObject(_))
+        .Times(2)
+        .WillRepeatedly(Return(&object1));
+    std::promise<bool> promise;
+    get_request_queue.AddRequest(
+        client, object_ids, /*timeout_ms*/ -1, false,
+        [&](const std::shared_ptr<GetRequest> &get_req) { promise.set_value(true); });
+    object1.state = ObjectState::PLASMA_SEALED;
+    get_request_queue.ObjectSealed(object_id);
+    promise.get_future().get();
+  }
+}
+}  // namespace plasma
```

Comment:
Can you also add a test that checks if the internal metadata fields are cleaned up properly? We can create a function `AssertNoLeak` and make sure at the end of each test suite, we make sure all of get request queue metadata is cleaned up.

Reply:
Done.